python/ray/util/sgd/tests/test_torch_runner.py

```diff
@@ -203,54 +203,96 @@ def _testWithInitialized(self, init_mock):
         os.environ["CUDA_VISIBLE_DEVICES"] = "0"
         with patch("torch.cuda.is_initialized") as init_mock:
             init_mock.return_value = True
-            self._testWithInitialized(init_mock)
+            self._testReserveCUDAResource(init_mock, 0)
 
-    def test2VisibleNotInitialized(self):
+    def test1VisibleWithInitializedAndReserveCPUResource(self):
+        os.environ["CUDA_VISIBLE_DEVICES"] = "0"
+        with patch("torch.cuda.is_initialized") as init_mock:
+            init_mock.return_value = True
+            self._testReserveCUDAResource(init_mock, 2)
+
+    def test2VisibleWithInitialized(self):
         os.environ["CUDA_VISIBLE_DEVICES"] = "2,3"
         with patch("torch.cuda.is_initialized") as init_mock:
-            init_mock.return_value = False
-            mock_runner = MagicMock()
-            mock_runner._set_cuda_device = MagicMock()
-            LocalDistributedRunner._try_reserve_resources_and_set_cuda(
-                mock_runner, 0, 1)
-            args, _ = mock_runner._set_cuda_device.call_args
-            self.assertTrue(("1" in args) or "0" in args)
-            self.assertEquals(len(os.environ["CUDA_VISIBLE_DEVICES"]), 1)
+            init_mock.return_value = True
+            self._testReserveCUDAResource(init_mock, 0)
+
+    def test2VisibleWithInitializedAndReserveCPUResource(self):
+        os.environ["CUDA_VISIBLE_DEVICES"] = "2,3"
+        with patch("torch.cuda.is_initialized") as init_mock:
+            init_mock.return_value = True
+            self._testReserveCUDAResource(init_mock, 2)
 
     def test1VisibleNotInitialized(self):
         os.environ["CUDA_VISIBLE_DEVICES"] = "0"
         with patch("torch.cuda.is_initialized") as init_mock:
             init_mock.return_value = False
-            mock_runner = MagicMock()
-            mock_runner._set_cuda_device = MagicMock()
-            LocalDistributedRunner._try_reserve_resources_and_set_cuda(
-                mock_runner, 0, 1)
-            mock_runner._set_cuda_device.assert_called_with("0")
-            self.assertEquals(len(os.environ["CUDA_VISIBLE_DEVICES"]), 1)
+            self._testReserveCUDAResource(init_mock, 0)
+
+    def test1VisibleNotInitializedAndReserveCPUResource(self):
+        os.environ["CUDA_VISIBLE_DEVICES"] = "0"
+        with patch("torch.cuda.is_initialized") as init_mock:
+            init_mock.return_value = False
+            self._testReserveCUDAResource(init_mock, 2)
+
+    def test2VisibleNotInitialized(self):
+        os.environ["CUDA_VISIBLE_DEVICES"] = "2,3"
+        with patch("torch.cuda.is_initialized") as init_mock:
+            init_mock.return_value = False
+            self._testReserveCUDAResource(init_mock, 0)
+
+    def test2VisibleNotInitializedAndReserveCPUResource(self):
+        os.environ["CUDA_VISIBLE_DEVICES"] = "2,3"
+        with patch("torch.cuda.is_initialized") as init_mock:
+            init_mock.return_value = False
+            self._testReserveCUDAResource(init_mock, 2)
```

Comment:
I don't think we need this test, it seems redundant

Reply:
removed